### ISSUE TITLE
data.sql is not working for initializing 'roles' table with prepopulated data

### ISSUE THREAD

**Issue by rajeshsinha1997** [rajeshsinha1997/Centralised_Banking_System_Application#6]
![image](https://user-images.githubusercontent.com/36392291/133896189-b9ef7977-bd0e-4fb1-a01a-94b41e41dc98.png)

When the application is starting, it should pick the data.sql from src/main/resource and execute that file. But that is not happening.

![image](https://user-images.githubusercontent.com/36392291/133896220-49175eb2-37e1-49fc-8886-b58c67d83c1f.png)

ddl-auto is set to 'create'

**Comment by osweet** [comment]
Hi @rajeshsinha1997. I can take a look at this issue. I cannot find the data-mysql.sql in the repository. Can you check if it is pushed on your side?

**Comment by rajeshsinha1997** [comment]
> Hi @rajeshsinha1997. I can take a look at this issue. I cannot find the data-mysql.sql in the repository. Can you check if it is pushed on your side?

Hi @osweet, the data-mysql.sql file is present at the 'src/main/resource' directory. If you're still not able to find it, please try to take the latest pull from the 'master' branch or clone the repository one more time. Hope this will solve the issue.

**Comment by osweet** [comment]
Yes, you are right. I cloned the repo, but I needed to pull after that as well. All is in now.

**Comment by rajeshsinha1997** [comment]
> Yes, you are right. I cloned the repo, but I needed to pull after that as well. All is in now.

That's good to hear!

**Comment by osweet** [comment]
@rajeshsinha1997 Do you have the DDL file - schema-mysql.sql? I can test the data-mysql.sql, but I need a database structure against which it will run. Can you send me the schema, or push it to the repo?

**Comment by rajeshsinha1997** [comment]
I've pushed both schema-mysql.sql and db_structure. You can find them at 'src/main/resource'

**Comment by osweet** [comment]
Thank you. I looked at the issue briefly tonight, plan to continue tomorrow

**Comment by osweet** [comment]
@rajeshsinha1997 Your scripts can easily be executed on a MySQL database, but cannot be executed by Spring without errors. The problem is in the foreign keys and in my opinion, Spring does something extra after it has created the tables and this extra thing leads to errors. Otherwise, I have configured it successfully to create the schema. I want to take a bit more time to check the issue so we can have working db schema and data loading.
If you want, I can push the changes up to now, but please let me know, as they are not finished yet.

**Comment by rajeshsinha1997** [comment]
@osweet, you can take your time as there is no hurry, and I'm thankful for the efforts you're giving to help me out with the issue.
Also just to give you a better explanation, with my entity classes the tables are getting created as they are supposed to be. My requirement is just to push some data to 'roles', 'users', and 'user_roles' tables once they are created.

**Comment by osweet** [comment]
@rajeshsinha1997 I have found the issue. It was configuration of application.properties - about where to find the data script, and the script itself - I rearranged two inserts, so they can work. Now the data is imported correctly.
I have also renamed the schema-mysql.sql script to a name, which will not be picked up by Spring. The reason is that when executing the schema-mysql.sql, Spring does not manage properly foreign keys. It creates the tables, but the execution exits.
I will make a pull request for you where you can see and review the changes.
